esphome/components/sprinkler/__init__.py

```diff
@@ -0,0 +1,624 @@
+    "sprinkler.previous_valve", PreviousValveAction, SPRINKLER_ACTION_SCHEMA
+)
+@automation.register_action("sprinkler.pause", PauseAction, SPRINKLER_ACTION_SCHEMA)
+@automation.register_action("sprinkler.resume", ResumeAction, SPRINKLER_ACTION_SCHEMA)
+@automation.register_action(
+    "sprinkler.resume_or_start_full_cycle", ResumeOrStartAction, SPRINKLER_ACTION_SCHEMA
+)
+@automation.register_action(
+    "sprinkler.shutdown", ShutdownAction, SPRINKLER_ACTION_SCHEMA
+)
+async def sprinkler_simple_action_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    return cg.new_Pvariable(action_id, template_arg, paren)
+
+
+async def to_code(config):
+    for sprinkler_controller in config:
+        if len(sprinkler_controller[CONF_VALVES]) > 1:
+            var = cg.new_Pvariable(
+                sprinkler_controller[CONF_ID],
+                sprinkler_controller[CONF_MAIN_SWITCH][CONF_NAME],
+            )
+        else:
+            var = cg.new_Pvariable(
+                sprinkler_controller[CONF_ID],
+                sprinkler_controller[CONF_VALVES][0][CONF_VALVE_SWITCH][CONF_NAME],
+            )
+        await cg.register_component(var, sprinkler_controller)
+
+        if len(sprinkler_controller[CONF_VALVES]) > 1:
+            sw_var = cg.new_Pvariable(sprinkler_controller[CONF_MAIN_SWITCH][CONF_ID])
+            await cg.register_component(sw_var, sprinkler_controller[CONF_MAIN_SWITCH])
+            await switch.register_switch(sw_var, sprinkler_controller[CONF_MAIN_SWITCH])
+            sw_var = await cg.get_variable(
+                sprinkler_controller[CONF_MAIN_SWITCH][CONF_ID]
+            )
+            cg.add(var.set_controller_main_switch(sw_var))
+
+            sw_aa_var = cg.new_Pvariable(
+                sprinkler_controller[CONF_AUTO_ADVANCE_SWITCH][CONF_ID]
+            )
+            await cg.register_component(
+                sw_aa_var, sprinkler_controller[CONF_AUTO_ADVANCE_SWITCH]
+            )
+            await switch.register_switch(
+                sw_aa_var, sprinkler_controller[CONF_AUTO_ADVANCE_SWITCH]
+            )
+            sw_aa_var = await cg.get_variable(
+                sprinkler_controller[CONF_AUTO_ADVANCE_SWITCH][CONF_ID]
+            )
+            cg.add(var.set_controller_auto_adv_switch(sw_aa_var))
+
+            if CONF_QUEUE_ENABLE_SWITCH in sprinkler_controller:
+                sw_qen_var = cg.new_Pvariable(
+                    sprinkler_controller[CONF_QUEUE_ENABLE_SWITCH][CONF_ID]
+                )
+                await cg.register_component(
+                    sw_qen_var, sprinkler_controller[CONF_QUEUE_ENABLE_SWITCH]
+                )
+                await switch.register_switch(
+                    sw_qen_var, sprinkler_controller[CONF_QUEUE_ENABLE_SWITCH]
+                )
+                sw_qen_var = await cg.get_variable(
+                    sprinkler_controller[CONF_QUEUE_ENABLE_SWITCH][CONF_ID]
+                )
```

Comment:
Tried something like this?
```suggestion
                sw_qen_var = switch.new_switch(sprinkler_controller[CONF_QUEUE_ENABLE_SWITCH])
```

Reply:
Nice...thanks again! Updated... 😄